File: day-3/part-1/elixir/test_power_consumption.exs
ExUnit.start()

defmodule PowerConsumptionTest do
  use ExUnit.Case

  test "calculate" do
    input = """
      00100
      11110
      10110
      10111
      10101
      01111
      00111
      11100
      10000
      11001
      00010
      01010
    """
    assert PowerConsumption.calculate(input) == 198
  end
end

File: day-3/part-1/ruby/test_power_consumption.rb
require_relative 'power_consumption'
require 'test/unit'

class TestPowerConsumption < Test::Unit::TestCase
  def test_calculate
    input = <<~TEXT
      00100
      11110
      10110
      10111
      10101
      01111
      00111
      11100
      10000
      11001
      00010
      01010
    TEXT

    assert_equal(198, PowerConsumption.new.calculate(input))
  end
end